Modules/User/Services/UserUpdateService.php

```diff
@@ -28,8 +28,7 @@ public function __construct(UserRepositoryInterface $userRepo) {
      * @return User|null
      */
     public function update(array $data) : ?User {
-        $this->userRepo->update($data, $data["id"]);
-        $user = $this->userRepo->findBy('id', $data["id"]);
+        $user = $this->userRepo->updateAndReturn($data, $data["id"]);
```

Comment:
puedes hacer directo 
`return $this->userRepo->updateAndReturn($data, $data["id"]);`